src/platform/packages/shared/kbn-unified-field-list/src/containers/unified_field_list_sidebar/field_list_sidebar_container.tsx

```diff
@@ -99,281 +222,214 @@ export type UnifiedFieldListSidebarContainerProps = Omit<
-                  id="unifiedFieldList.fieldListSidebar.fieldsMobileButtonLabel"
-                  defaultMessage="Fields"
-                />
-                <EuiBadge
-                  className="unifiedFieldListSidebar__mobileBadge"
-                  color={workspaceSelectedFieldNames?.[0] === '_source' ? 'default' : 'accent'}
-                >
-                  {!workspaceSelectedFieldNames?.length ||
-                  workspaceSelectedFieldNames[0] === '_source'
-                    ? 0
-                    : workspaceSelectedFieldNames.length}
-                </EuiBadge>
-              </EuiButton>
-            </div>
-            {isFieldListFlyoutVisible && (
-              <EuiPortal>
-                <EuiFlyout
-                  size="s"
-                  onClose={() => setIsFieldListFlyoutVisible(false)}
-                  aria-labelledby="flyoutTitle"
-                  ownFocus
-                >
-                  <EuiFlyoutHeader hasBorder>
-                    <EuiTitle size="s">
-                      <h2 id="flyoutTitle">
-                        <EuiLink color="text" onClick={() => setIsFieldListFlyoutVisible(false)}>
-                          <EuiIcon
-                            className="eui-alignBaseline"
-                            aria-label={i18n.translate(
-                              'unifiedFieldList.fieldListSidebar.flyoutBackIcon',
-                              {
-                                defaultMessage: 'Back',
-                              }
-                            )}
-                            type="arrowLeft"
-                          />{' '}
-                          <strong>
-                            {i18n.translate('unifiedFieldList.fieldListSidebar.flyoutHeading', {
-                              defaultMessage: 'Field list',
-                            })}
-                          </strong>
-                        </EuiLink>
-                      </h2>
-                    </EuiTitle>
-                  </EuiFlyoutHeader>
-                  <UnifiedFieldListSidebar
-                    {...commonSidebarProps}
-                    alwaysShowActionButton={true}
-                    buttonAddFieldVariant="primary" // always for the flyout
-                    isSidebarCollapsed={undefined}
-                    prepend={prependInFlyout?.()}
-                  />
-                </EuiFlyout>
-              </EuiPortal>
-            )}
-          </>
-        );
-      };
-
-      if (variant === 'button-and-flyout-always') {
-        return renderButtonVariant();
+  useEffect(() => {
+    const cleanup = () => {
+      if (closeFieldEditor?.current) {
+        closeFieldEditor?.current();
       }
+    };
+    return () => {
+      // Make sure to close the editor when unmounting
+      cleanup();
+    };
+  }, []);
 
-      if (variant === 'list-always') {
-        return renderListVariant();
-      }
+  useImperativeHandle(
+    componentRef,
```

Comment:
Had to split with component in 2 as it's defining the ref API too.

Before:
```
<UnifiedFieldListSidebarContainer with memo />
```

After:
```
const UnifiedFieldListSidebarContainerWithRestorableState = withRestorableState(
  memo(InternalUnifiedFieldListSidebarContainer)
)
// and
<UnifiedFieldListSidebarContainer>
  <UnifiedFieldListSidebarContainerWithRestorableState />
</UnifiedFieldListSidebarContainer>
```